Source/Core/DolphinQt2/GameList/GameFile.cpp

```diff
@@ -21,6 +23,16 @@
 static const int CACHE_VERSION = 13;  // Last changed in PR #3261
 static const int DATASTREAM_VERSION = QDataStream::Qt_5_5;
 
+u64 GetTitleID(const QString& path)
+{
+  const auto volume = DiscIO::CreateVolumeFromFilename(path.toStdString());
+
+  u64 title_id;
+  volume->GetTitleID(&title_id);
```

Comment:
Is there a reason why you're not handling the case where `CreateVolumeFromFilename` returns `nullptr`?